clientconn.go

```diff
@@ -170,17 +170,17 @@ func WithInsecure() DialOption {
 
 // WithTransportCredentials returns a DialOption which configures a
 // connection level security credentials (e.g., TLS/SSL).
-func WithTransportCredentials(creds credentials.TransportAuthenticator) DialOption {
+func WithTransportCredentials(auth credentials.TransportAuthenticator) DialOption {
```

Comment:
Thanks for your review.
The reason for not change the function name is this is a public api, and we are trying to minimize breakage.

How about changing the interface name from `TransportAuthenticator` to `TransportCredentials`? So this function will look like
```
func WithTransportCredentials(creds credentials.TransportCredentials) DialOption
```